Data/Vector/Generic.hs

```diff
@@ -2205,9 +2199,19 @@ gfoldl :: (Vector v a, Data a)
 {-# INLINE gfoldl #-}
 gfoldl f z v = z fromList `f` toList v
 
-mkType :: String -> DataType
-{-# INLINE mkType #-}
-mkType = mkNoRepType
+mkTypeConstr :: String -> (DataType, Constr)
```

Comment:
Tupling these together feels a mite strange. Why not separate these into two separate functions which return `DataType` and `Constr` respectively? Then you wouldn't have to manually project out of it with `fst` and `snd` everywhere.

Reply:
I only tupled them because they're mutually recursive, but I know that's a bit of a silly reason. I'll separate them this evening when I get a chance.